### PR TITLE
retrying on CRUD API calls

### PR DESCRIPTION
**Pull request by a6sondhi**
By retrying on Create, Update and Delete operations, we are able to
avoid failures due to the order of the operation. (i.e. trying to create
a group with the same name before an update)

### CODE REVIEW

File: okta/resource_group.go [hunk 1]
```diff
@@ -86,23 +93,31 @@ func resourceGroupRead(d *schema.ResourceData, m interface{}) error {
 }
 
 func resourceGroupUpdate(d *schema.ResourceData, m interface{}) error {
-	group := buildGroup(d)
-	_, _, err := getOktaClientFromMetadata(m).Group.UpdateGroup(d.Id(), *group)
-	if err != nil {
-		return err
-	}
+	return resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
```
Comment (badie) [inline okta/resource_group.go]: use `schema.TimeoutUpdate` instead for this operation

https://github.com/hashicorp/terraform/blob/master/helper/schema/resource_timeout.go#L19

File: okta/resource_group.go [hunk 2]
```diff
@@ -86,23 +93,31 @@ func resourceGroupRead(d *schema.ResourceData, m interface{}) error {
...
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
 }
 
 func resourceGroupDelete(d *schema.ResourceData, m interface{}) error {
-	_, err := getOktaClientFromMetadata(m).Group.DeleteGroup(d.Id())
+	return resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
```
Comment (badie) [inline okta/resource_group.go]: use `schema.TimeoutDelete` instead for this operation

https://github.com/hashicorp/terraform/blob/master/helper/schema/resource_timeout.go#L20

File: okta/resource_group.go (hunk 1 above)
Comment (a6sondhi) [reply okta/resource_group.go]: nice catch, copy pasta haha

**Comment by a6sondhi** [comment]
@badie, merge this as required, I don't have access to do that anymore.

**Comment by ebabani** [comment]
This is no longer required.

**Pull request closed without merging**